extensions/amp-ad-exit/0.1/amp-ad-exit.js

```diff
@@ -0,0 +1,200 @@
+import {urlReplacementsForDoc} from '../../../src/services';
+import {user} from '../../../src/log';
+
+const TAG = 'amp-ad-exit';
+
+export class AmpAdExit extends AMP.BaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {!./config.AmpAdExitConfig} */
+    this.config_ = {targets: {}, filters: {}};
+
+    this.defaultFilterSpecs_ = {
+      'minDelay': makeClickDelaySpec(1000),
+    };
+
+    this.clickDelayFilter_ = new ClickDelayFilter();
+
+    this.registerAction('exit', this.exit.bind(this));
+  }
+
+  /**
+   * @param {!../../../src/service/action-impl.ActionInvocation} invocation
+   */
+  exit({args, event}) {
+    if (!isExperimentOn(this.win, 'amp-ad-exit')) {
+      user().warn('amp-ad-exit experiment is off.');
+      return;
+    }
+
+    event.preventDefault();
+    const targetName = args.target;
+    const target = this.config_.targets[targetName];
+    if (!target) {
+      user().error(TAG, `Exit target not found: '${targetName}'`);
+      return;
+    }
+    if (!this.filter_(
+            Object.keys(this.defaultFilterSpecs_), this.defaultFilterSpecs_,
+            event) ||
+        !this.filter_(target.filters, this.config_.filters, event)) {
+      return;
+    }
+    const substituteVariables =
+        this.getUrlVariableRewriter_(args, event, target);
+    if (target.trackingUrls) {
+      target.trackingUrls.map(substituteVariables)
+          .forEach(url => this.pingTrackingUrl_(url));
+    }
+    openWindowDialog(this.win, substituteVariables(target.finalUrl), '_blank');
+  }
+
+
+  /**
+   * @return {function(string): string}
+   */
+  getUrlVariableRewriter_(args, event, target) {
+    const vars = {
+      'CLICK_X': () => event.clientX,
+      'CLICK_Y': () => event.clientY,
+    };
+    const whitelist = {
+      'RANDOM': true,
+      'CLICK_X': true,
+      'CLICK_Y': true,
+    };
+    if (target.vars) {
+      for (const customVar in target.vars) {
+        if (customVar.startsWith('_')) {
+          vars[customVar] = () =>
+              args[customVar] || target.vars[customVar].defaultValue;
+          whitelist[customVar] = true;
+        }
+      }
+    }
+    const replacements = urlReplacementsForDoc(this.getAmpDoc());
+    return url => replacements.expandUrlSync(
+        url, vars, undefined /* opt_collectVars */, whitelist);
+  }
+
+  /**
+   * Attemts to issue a request to `url` to report the click. The request method
+   * depends on the exit config's transport property. navigator.sendBeacon will
+   * be tried if transport.beacon is `true` or `undefined`. Otherwise, or if
+   * sendBeacon returns false, an image request will be made.
+   * @param {string} url
+   */
+  pingTrackingUrl_(url) {
+    user().fine(TAG, `pinging ${url}`);
+    const useBeacon = this.config_.transport[TransportMode.BEACON] == true ||
+        !this.config_.transport.hasOwnProperty(TransportMode.BEACON);
```

Comment:
This can be simplified to `this.config_.transport[TransportMode.BEACON] !== false`, once you fix the above error.